src/Val.cc

```diff
@@ -3001,7 +3022,20 @@ void RecordVal::DoneParsing()
 	parse_time_records.clear();
 	}
 
-const ValPtr& RecordVal::GetField(const char* field) const
+Val* RecordVal::Lookup(int field) const
```

Comment:
Right, I was hoping that the leak is okay given it's deprecated, as I don't see any way to fix it.  The `GetField` method behaves appropriately (no leak).

Reply:
I have doubts about how acceptable it is to introduce a leak into now-deprecated code -- seems like just another shade of "we broke this API" and I'd maybe even prefer a build-time breakage due to removing it rather than a run-time misbehavior.

I haven't finished paging in all the details of the PR and about to run out of time for a while, but if the "is managed" property of `ZamValUnion` were per-object rather than based on the static list of managed types, maybe it could do something like convert the unmanaged value into a managed one at the point when someone is found using the deprecated API (but don't know how much other difficulties there are with changing everything to now be like a `struct { ZAMValUnion zval; bool is_managed; }`... plus maybe that's just another form of begging to go down the road of wrapping things into a more concise representation that's type/memory safe like `variant`... which was an initial impression I share with Robin if there's desire to do more brainstorming toward making that wishlist-item come true).

Generally, if there's other ideas for having temporary slow-paths or ways to opt-in to breakages (compile-time options?), that could be a way to address things like this -- any such workaround would only exist in `master` until 4.0 is released (planned for around end of this year) and then they'd get removed along with the deprecations.

Reply:
I've pushed a simpler change that tracks ValPtr's returned by the deprecated `Lookup()` method and recovers the storage upon destruction of the `RecordVal`.  I tested this using the `#if` code I introduced for `GetField()`, which is commented out in the pushed version.